### PR TITLE
feat(APIM-151): global enum introduction

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction
Lack of products related constants.

## Resolution
* Global enum introduction and consumption.
* `PRODUCTS` constants introduction and consumption.
* Enhanced test coverage.

## Miscellaneous
* `.env.sample` comments

### CODE REVIEW

File: src/modules/exposure-period/dto/get-exposure-period-query.dto.ts [hunk 1]
```diff
@@ -12,6 +13,6 @@ export class GetExposurePeriodQueryDto {
 
   @IsString()
   @ApiProperty({ example: 'EW', description: 'Two products are accepted: EW and BS' })
-  @Matches(/^(EW|BS)$/)
+  @IsEnum(ENUMS.PRODUCTS)
```
Comment (IgnatG) [inline src/modules/exposure-period/dto/get-exposure-period-query.dto.ts:16]: does this work? the `@Matches` validator may not translate directly into `@IsEnum`

File: test/exposure-period/exposure-period.api-test.ts [hunk 2]
```diff
@@ -101,23 +102,23 @@ describe('Exposure period', () => {
...
     expect(body.message).toContain('enddate must be a Date instance');
     expect(body.message).toContain('productgroup must match /^(EW|BS)$/ regular expression');
   });
 
   it('GET /exposure-period?startdate=undefined&enddate=undefined&productgroup=undefined', async () => {
-    const { status, body } = await api.get('/exposure-period?startdate=undefined&enddate=undefined&productgroup=undefined');
+    const { status, body } = await api.get(`/exposure-period?startdate=undefined&enddate=undefined&productgroup=undefined`);
     expect(status).toBe(400);
     expect(body.message).toContain('startdate must be a Date instance');
     expect(body.message).toContain('enddate must be a Date instance');
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply test/exposure-period/exposure-period.api-test.ts:124]: Good spot, have rectified.

File: src/modules/exposure-period/dto/get-exposure-period-query.dto.ts (hunk 1 above)
Comment (abhi-markan) [reply src/modules/exposure-period/dto/get-exposure-period-query.dto.ts:16]: Yep this does work.

File: test/premium-schedules/premium-schedules.api-test.ts [hunk 3]
```diff
@@ -46,7 +47,7 @@ describe('Premium schedules', () => {
     const createSchedules = [
       {
         facilityURN: chance.natural({ min: 10000000, max: 99999999 }),
-        productGroup: 'BS',
+        productGroup: PRODUCTS.BS,
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply test/premium-schedules/premium-schedules.api-test.ts:50]: Can you try again please.

[the remark this answers is not in the document]
Comment (avaitonis) [reply test/premium-schedules/premium-schedules.api-test.ts:50]: Still fails with error "productGroup must be one of the following values: [object Object]"

File: src/modules/exposure-period/dto/get-exposure-period-query.dto.ts (hunk 1 above)
Comment (avaitonis) [reply src/modules/exposure-period/dto/get-exposure-period-query.dto.ts:16]: ENUM.PRODUCTS contain object { QueryParamProductsEnum: { EW: 'EW', BS: 'BS' } } . 
Enum has to be spread for ENUMS.PRODUCTS  or referenced using ENUMS.PRODUCTS.QueryParamProductsEnum

File: test/premium-schedules/premium-schedules.api-test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (abhi-markan) [reply test/premium-schedules/premium-schedules.api-test.ts:50]: Have rectified, accompanied with enhanced test coverage.

File: src/modules/exposure-period/dto/get-exposure-period-query.dto.ts (hunk 1 above)
Comment (abhi-markan) [reply src/modules/exposure-period/dto/get-exposure-period-query.dto.ts:15]: Have rectified, accompanied with enhanced test coverage.

**Review by avaitonis — APPROVED** [review]
LGTM

**Pull request merged**